### ISSUE TITLE
Content and encoding of person "minister" function 

### ISSUE THREAD

**Issue by TomazErjavec** [clarin-eric/ParlaMint#185]
In ParlaMint II we will also need to add "information about members of the governments (ministers)". This issue is to discuss how to encode this information, as well as what exactly this information is, as the above statement is somewhat vague. 
As to what information we would encode:

1. only the fact that somebody is/was a minister and from-to when
2. additonaly also what they were a minister of, which means that we would need to introduce also the ministries into (probably) organisations, also marked with temporal attributes, as the ministries can change after a new gov. comes to power
3. also other prominent functions, in particular the primer minister and/or president

The encoding would be within the [person](https://www.tei-c.org/release/doc/tei-p5-doc/en/html/ref-person.html) elements most likely using the [state](https://www.tei-c.org/release/doc/tei-p5-doc/en/html/ref-state.html) element. The reason `state` is used instead of e.g. [occupation](https://www.tei-c.org/release/doc/tei-p5-doc/en/html/ref-occupation.html) is that the latter has text content, which is bad for encoding multilingual content; also, `state` can also be used for party orientation, cf. #184.

```
<state type="minister" ref="#ministerOfInterior" from="2022-01-15"> 
  <label>Minister of the Interior</label>
</state>
```
* `@type` is obligatory, and, if we go for option 2 above, so is `@ref` (which points to the ministry org)
*  state should be allowed the standard temporal attributes
*  label is optional, and we can have several for bi- or multilingual descriptions (of course, all except possibly one then need to be marked with `@xml:lang`).

@matyaskopp, would you agree?

**Comment by matyaskopp** [comment]
I like the idea of adding information about ministers (CZ already has it in a vague way - affiliation to the government without a more specific role). But I don't understand why are you introducing a new encoding for it. We already have `affiliation` that can handle it.
But we can look at the minister's situation from two sides:

1. minister **in government**
2. minister **in ministry**

We can probably say that every minister is a member of government:
```XML
  <affiliation role="member" ref="#government" ana="#government.t1" />
```
and is a minister in some ministry (except from Prime Minister):
```XML
  <affiliation role="minister" ref="#ministryOfHealth" />
```
The second affiliation produces one-man organizations without any benefit. Every country will definitely use their own names of ministry and it would not be possible to compare one minister type across countries.

I am suggesting to create a taxonomy for main/common ministry and add attribute type to the affiliation that would extend the role:
```XML
  <affiliation role="minister" type="HEALTH" ref="#government" ana="#government.t1" />
```
If there is a country specific ministry we will just use:
```XML
  <affiliation role="minister" ref="#government" ana="#government.t1" />
```

**Comment by TomazErjavec** [comment]
> I don't understand why are you introducing a new encoding for it. 

My bad; I forgot about affiliation, and of course agree that we should continue using it.

> We can probably say that every minister is a member of government

Hm, but surely they are not the only members of the government? Aren't MPs also (or at least those in coalition)? This would not tell you somebody is a minister then. 

> The second affiliation produces one-man organizations without any benefit.

I am not sure that it has no benefit:
* researchers might be interested only in one corpus, and here they will have the extra information who is a minster of what
* if the ministry `org` were connected to a taxonomy as you propose (and it sounds like a very nice idea), then you can do the cross-corpus comparisons. This would also be better than connecting to a taxonomy from the person affiliation via `@type`, which is not an IDREF and so has not connection to a taxonomy - the proper place for type values would be the schema, but we don't want to put it there.

In short, I propose (not sure about `@ana` vs. `@ref` of course):
```
<affiliation role="minister" ref="#ministryOfHealthAndSports-SI" />
...
<org xml:id="ministryOfHealthAndSports-SI" ana="#ministryHealth #ministrySport">
   <orgName xml:lang="en" full="yes">Ministry of Health and Sports</orgName>
</org>
```

**Comment by matyaskopp** [comment]
> Hm, but surely they are not the only members of the government? 

No, from the Czech perspective (@katjameden can you support my claim?)
We have only ministers and prime minister in government - it can be different people from MP, they shouldn't be a MP.

> Aren't MPs also (or at least those in coalition)? This would not tell you somebody is a minister then.

It is called "government political party". It is more like a relation between a political party(not its members) and government.
We have encoded that a party/group is a "government political party" when it is in a coalition. In case there is only one government political party, we don't encode it (we don't have one party coalition)

> I am not sure that it has no benefit:

Ok, if we add additional annotation of type of ministry, it makes sense.

> In short, I propose (not sure about @ana vs. @ref of course):
```
<affiliation role="minister" ref="#ministryOfHealthAndSports-SI" />
...
<org xml:id="ministryOfHealthAndSports-SI" ana="#ministryHealth #ministrySport">
   <orgName xml:lang="en" full="yes">Ministry of Health and Sports</orgName>
</org>
```

I think that encoding is ok in this way.

Great, I like the idea of annotating topics, not ministry names. We have sports affiliated with the ministry of education.

```XML
<org xml:id="ministryOfEducationAndSports-CZ" ana="#ministryEducation #ministrySport">
   <orgName xml:lang="en" full="yes">Ministry of Education, Youth and Sports</orgName>
</org>
```

So we(@katjameden ?) should list topics and then create a taxonomy. If we agree on that, we can make a new issue.

**Comment by katjameden** [comment]
> > Hm, but surely they are not the only members of the government?
> 
> No, from the Czech perspective (@katjameden can you support my claim?) We have only ministers and prime minister in government - it can be different people from MP, they shouldn't be a MP.

I am not an expert on this, but I asked my collegue for clarification: 

Both for the Slovene and the Czech cases, the perspective is the same: MPs cannot be Ministers (in the case where one is an MP and is appointed minister, one has to resign as an MP). The reason why ministers are sometimes speakers in the Parliament is the fact, that they have to present/defend their work in front of MPs (ministers are the members of the executive political branch and parliament represents legislative branch and in every liberal democracy these have to be separated).
> 
> > Aren't MPs also (or at least those in coalition)? This would not tell you somebody is a minister then.
> 
> It is called "government political party". It is more like a relation between a political party(not its members) and government. We have encoded that a party/group is a "government political party" when it is in a coalition. In case there is only one government political party, we don't encode it (we don't have one party coalition).
> I think that encoding is ok in this way.
> 
> Great, I like the idea of annotating topics, not ministry names. We have sports affiliated with the ministry of education.
> 
> ```
> <org xml:id="ministryOfEducationAndSports-CZ" ana="#ministryEducation #ministrySport">
>    <orgName xml:lang="en" full="yes">Ministry of Education, Youth and Sports</orgName>
> </org>
> ```
> 
> So we(@katjameden ?) should list topics and then create a taxonomy. If we agree on that, we can make a new issue.

Yes, that would be great.

**Comment by matyaskopp** [comment]
>  MPs cannot be Ministers (in the case where one is an MP and is appointed minister, one has to resign as an MP). 

No, the Czech case is that being the minister does not imply being a MP but it is possible that the minister is also MP :-(.
BTW: this fact brings confusion to czech data, stenographers do not distinguish if the person is speaking from the position of MP or minister - they always assign minister role to the speaker (if the speaker is also a minister) and don't care if speaker is speaking as MP or minister.

> Yes, that would be great.

moving ministry taxonomy discussion to #190

**Comment by coltekin** [comment]
Since I am working on getting the minister information included, I want to check if I understand the current proposal. The part I am not fully certain is do we plan to have an organization per ministry, or per 'government'.

To me, per-government approach is more attractive, since  - at least in Turkey - there are/were some ministers called "state ministers" who actually do not seem to have a well-defined "minister" role. If the organization is the government or cabinet, then we also do not need to single out the prime minister. He/she can simply be another person affiliated with the government. Similarly, in some countries, head of the government may be the president, or maybe named in a different way. Similarly, vice president or deputy (prime) ministers may also be frequent speakers.

If we follow this, one approach could be to define an organization per cabinet (term of the government), or alternatively, a single organization with `<event>`s or other temporal objects associated with it (similar to parliament terms).

While discussing this with a colleague, I also learned that in some countries (at least in UK), there may be ministers outside the cabinet too. So, covering all possible cases maybe difficult.

In any case, I like the idea of building a taxonomy, but I think we also may need to be flexible. I expect there will be rather creative minister roles.

**Comment by matyaskopp** [comment]
> Since I am working on getting the minister information included, I want to check if I understand the current proposal. The part I am not fully certain is do we plan to have an organization per ministry, or per 'government'.

> To me, per-government approach is more attractive, since - at least in Turkey - there are/were some ministers called "state ministers" who actually do not seem to have a well-defined "minister" role. If the organization is the government or cabinet, then we also do not need to single out the prime minister. He/she can simply be another person affiliated with the government. Similarly, in some countries, head of the government may be the president, or maybe named in a different way. Similarly, vice president or deputy (prime) ministers may also be frequent speakers.

I think we should have both:
- grouping ministers in **government** (or cabinet) tells us that they are working together in government and they have a boss  **Prime Minister**(premier, president, chancellor, ...) and possibly  some **state ministers**
- affiliating with **ministry** organization (one `org` for each ministry) tells us what is the responsibility of the minister.

It is a good point - How should we call the Prime Minister role - shouldn't it be something more general instead of using British naming? 

> If we follow this, one approach could be to define an organization per cabinet (term of the government), or alternatively, a single organization with <event>s or other temporal objects associated with it (similar to parliament terms).

We already have `<event>` encoding of governments (note that the timespans and number of governments shouldn't be necessarily equal to terms). 
This simplification does not allow adding events related to one government (government in demission, motion of no confidence, motion of confidence) but I think we can keep it as is.

Probably we need an opinion from a more west political system - the former Eastern Bloc of USSR, former Yugoslavia and Turkey is not a representative sample :-)

**Comment by matyaskopp** [comment]
I think it can be done this way:
```mermaid
graph LR
    PM[fa:fa-user Prime Minister] --primeMinister--> G(Government):::pw
    SM[fa:fa-user State Minister] --stateMinister--> G
    MH[fa:fa-user Minister of Health] --minister--> G
    DM[fa:fa-user Deputy Minister of Health] -.deputyMinister.-> G
    MH & DM --minister--> H(Ministry of Health):::pw
    MH & PM & SM & DM -.MP...-> P(Parliament)
    linkStyle 3 stroke:#69f,color:ref;
    classDef pw fill:#f96;
```
where:
- every person with `<affiliation role="minister" ...>` should be affiliated both with **government** and his own **ministry**
- other ministers like persons (Prime Minister, State Minister,...) that don't have their ministry should be affiliated only with **government**
- I am not quite sure if other persons that are related to specific **ministry** should also be in government (blue line) - I am leaning towards affiliating them only with the ministry (to remove blue arrow)
- all persons can be affiliated with parliament if they are in parliament (allowed in CZ, not allowed in SI)

@coltekin @katjameden @TomazErjavec  Do we agree on this?

If yes, this needed to be done:
- extend validation: `minister` role affiliated to government implies custom ministry organization 
- add ministry to organization role in odd: https://github.com/clarin-eric/ParlaMint/blob/0c9fed81a5a417b92f6e3f5b1ec8a41e9da1a1c6/TEI/ParlaMint-schemaSpecs.editing.odd.xml#L1360
- add ministry to organization role in documentation + describe it in text: https://github.com/clarin-eric/ParlaMint/blob/0c9fed81a5a417b92f6e3f5b1ec8a41e9da1a1c6/TEI/ParlaMint.odd.xml#L1251-L1261
- add ministry to schema: https://github.com/clarin-eric/ParlaMint/blob/57050d207444ccb051bf46b7ec44926f96f9e73a/Schema/ParlaMint-teiCorpus.rng#L189

**Comment by coltekin** [comment]
I fully agree with this setup - including not associating the "deputy ministers" with the government directly, but through the ministries.

**Comment by TomazErjavec** [comment]
I sortof agree, but not for everything, here are my comments:
* We should definitely decouple the minister function from being an MP, as this obviously differs between countries - in SI a minister cannot be an MP, but in GB (from a separate email thread, info by Paul Rayson) they can, and it seems in CZ too. So, this is ok in the diagram.
* There can be ministers that do not have a ministry, in particular [ministers without portfolio](https://en.wikipedia.org/wiki/Minister_without_portfolio), and we have such [in Slovenia](https://www.gov.si/en/state-authorities/government/about-the-government/). So, I would not insist that "every person with `<affiliation role="minister" ...>` should be affiliated both with government and his own ministry". Rather, they *should* be only affiliated with the government but *can* be affiliated with a ministry. This would also give the escape hatch for the case when we don't actually manage to set up the whole ministry organisation and taxonomy infrastructure for all corpora, i.e. we would still know somebody is a minister, only not of what.
* I am not sure if we want the same label "minister" for the affiliation that points to governemnt and to the ministry org, maybe the second one should be "ministerOf". However, I have problems formulating what bothers me with having the same label...
* I actually like the blue line, because, again, we can say somebody is a deputyMinister without saying of what; also, if there are deputyMinisters without portfolio, we would still be able to say that they exist. But note that we never promised to actually encode deputy ministers, just ministers.

**Comment by matyaskopp** [comment]
> - I am not sure if we want the same label "minister" for the affiliation that points to governemnt and to the ministry org, maybe the second one should be "ministerOf". However, I have problems formulating what bothers me with having the same label...

so `ministerOf` role affiliation with ministry would imply `minister` role affiliation with the government. Yes, this can be the solution. Previously I wanted to reserve `minister` role just for persons with the ministry. But this solution sounds less confusing

> - I actually like the blue line, because, again, we can say somebody is a deputyMinister without saying of what; also, if there are deputyMinisters without portfolio, we would still be able to say that they exist. But note that we never promised to actually encode deputy ministers, just ministers.

Yes, we did not promise to encode other roles. But in case the person with such affiliation has a speech in parliament and is related to ministry - we should be able to store such information.

**Comment by lnnoemi** [comment]
@TomazErjavec, I'm sorry, but I lost track. 
What is the final decision on the ministers? Do they belong to the government or the ministry? 
Or both? (If they are not ministers without portfolio)
Did you end up with a ministerOf role?

We now refer ministers to the government, as in Hungary we also have ministers without portfolio, thus we can encode all ministers the same way. But somewhere above you mentioned the possibility of referring a minister to both the government and its ministry.

Thanks!

**Comment by TomazErjavec** [comment]
> What is the final decision on the ministers? Do they belong to the government or the ministry? Or both? (If they are not ministers without portfolio) Did you end up with a ministerOf role?

We did not - at least so far - implement the ministerOf role (which would tie a person to an organisation denoting a particular ministry) because looking at the old ParlaMint corpora, I didn't see any corpora that actually encode ministries. If you (or anybody else!) wants to do that, I can certainly add this role and give an example in the guidelines.

So, for now, they are associated only the the gov'mnt organisation, and the text inside the affiliation can further specify what they are minister of, cf. the [guidelines](https://clarin-eric.github.io/ParlaMint/#sec-affiliation).

**Comment by lnnoemi** [comment]
Yes, I get it, thanks.
I do not see a `ministry` role among valid organisation roles - how should we encode ministries in the corpus?
I thought `<affiliation role="ministry" ...>` would be okay...
And sorry if this has been already discussed somewhere, I could not find it in the issues, my bad.

**Comment by TomazErjavec** [comment]
> I do not see a ministry role among valid organisation roles - how should we encode ministries in the corpus?

Well, as we don't have ministerOf, I thought we don't need the ministries either. 
However, you are probably not the only one that would encode this, so I fixed the schema and guidelines to allow ministry role on organisations. Also wanted to add ministerOf but then reconsidered, as I think we don't really need it, as we have head. So, a person that is affiliated with a ministry organisation with the role head, is the minister. @matyaskopp, do you agree?

These changes are currenlty only in the documentation branch, and also in the published Guidelines in the sections on
[Organisations](https://clarin-eric.github.io/ParlaMint/#sec-orgs) and [Speaker affiliations](https://clarin-eric.github.io/ParlaMint/#sec-affiliation).

So, with this, and if it seems all ok, we could even close this issue?

**Comment by matyaskopp** [comment]
> So, a person that is affiliated with a ministry organisation with the role head, is the minister. @matyaskopp, do you agree?

Agree. 

summary of affiliations:
```mermaid
graph LR
    MH[fa:fa-user Minister of Health] --minister--> G(Government -- role=government):::pw
    MH --member--> G
    MH  --head--> H(Ministry of Health -- role=ministry):::pw
    MH  --member--> H
    classDef pw fill:#f96;
```

**Comment by TomazErjavec** [comment]
Moving this to "future" milestone; the way it is in 3.1 is that those corpora that "natively" encoded ministers and ministries do it the way they do it. Those that have the ministers encoded as extra metadata in TSV have them added and encoded as e.g.
```
<affiliation role="minister" from="2012-01-12" to="2015-03-31"
             source="https://en.wikipedia.org/wiki/Ministry_of_Finance_and_Treasury_(Bosnia_and_Herzegovina)">
  <orgName full="yes">Ministarstvo finansija i trezora Bosne i Hercegovine</orgName>
  <orgName full="yes" xml:lang="en">Ministry of Finance and Treasury</orgName>
  <note type="period">11th cabinet</note>
</affiliation>
```
